ecmascript/transforms/optimization/src/simplify/branch/mod.rs

```diff
@@ -851,6 +851,10 @@ impl Fold for Remover {
             }
 
             Stmt::DoWhile(s) => {
+                if has_conditional_stopper(&[Stmt::DoWhile(s.clone())]) {
```

Comment:
This seems like an optimization is possible, but I'll invert the pr with https://github.com/swc-project/swc/pull/1613. Thanks!